src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs

```diff
@@ -283,12 +274,44 @@ private IEnumerable<ITypeSymbol> InferTypesWorker(int position)
                     var parentAssignment = token.Parent.Parent as AssignmentExpressionSyntax;
                     if (token.Parent == parentAssignment.Left)
                     {
-                        var grandparentAssignment = parentAssignment.Parent as AssignmentExpressionSyntax;
-                        if (grandparentAssignment.IsKind(SyntaxKind.SimpleAssignmentExpression) &&
-                            parentAssignment == grandparentAssignment.Right)
+                        // case 1:
+                        //      new Form
+                        //      {
+                        //          Location = new $$
+                        //          StartPosition = FormStartPosition.CenterParent
+                        //      };
+                        // The 'new' token is part of an assignment of the assignment to StartPosition,
+                        // but the user is really trying to assign to Location.
+                        // In this case, we need to detect that the 'new' token is on the left side of 
+                        // its parent assignment and on the right side of its grandparent assignment and
+                        // return the inferred type of the grandparent.
+                        if (parentAssignment.Parent is AssignmentExpressionSyntax)
                         {
-                            parent = grandparentAssignment;
-                            token = grandparentAssignment.OperatorToken;
+                            var grandparentAssignment = (AssignmentExpressionSyntax)parentAssignment.Parent;
+                            if (grandparentAssignment.IsKind(SyntaxKind.SimpleAssignmentExpression) &&
```

Comment:
Yes, good catch! :clap: I'd indeed forgotten about this. 

Haven't tested yet but some notes off the top of my mind:
The first check however is scoped for cases like below, which cannot have compound assignment :
```C#
Program p = new $$
bool b = false;
```

The second check for equals value clause though, should handle this. I'll test and address it :smile: 

Thanks again, good find.